opennms-provision/opennms-detector-simple/src/main/java/org/opennms/netmgt/provision/detector/snmp/SnmpDetector.java

```diff
@@ -220,6 +221,27 @@ public boolean isServiceDetected(InetAddress address, SnmpAgentConfig agentConfi
         }
     }
 
+    @Override
+    public List<SnmpAgentConfig> getListOfAgentConfigs(DetectRequest request) {
+        List<SnmpAgentConfig> agentConfigList = new ArrayList<>();
+        Map<String, String> runTimeAttributes = request.getRuntimeAttributes();
+        if (hasMultipleAgentConfigs(runTimeAttributes)) {
+            runTimeAttributes.remove(HAS_MULTIPLE_AGENT_CONFIGS);
+            //Rest of the attribute values are snmp configs.
+            runTimeAttributes.forEach((label, configAsString) -> {
+                agentConfigList.add(SnmpAgentConfig.parseProtocolConfigurationString(configAsString));
+            });
+        }
+        return agentConfigList;
+    }
+
+    @Override
+    public boolean isServiceDetected(InetAddress address, List<SnmpAgentConfig> agentConfigList) {
+        return agentConfigList.parallelStream().anyMatch(snmpAgentConfig -> {
```

Comment:
Since `SnmpUtils.getOids` and `SnmpUtils.getOid` are static and isServiceDetected doesn't update any state, I think it is safe to use `parallelStream` here.  

Note that since we are updating the profiles list into map and getting back them from map, the order of snmp profiles is not preserved.

Reply:
We shouldn't use `parallelStream()` with the default thread pool for blocking operations like these.